### PR TITLE
add commonjs export keyword

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Adds `woof` as a keyword for `module.exports`

### CODE REVIEW

File: LANGUAGE.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Should dogescript be ported to other languages, the `js` portion may be changed
 
 * dogescript uses single quotes for strings. Double quotes are not supported.
 * dogescript uses 4 space characters for indentation. Tabs are not supported.
-* dogescript seperates statements by newlines by default. In true-doge mode, they are separated by 3 spaces.
+* dogescript separates statements by newlines by default. In true-doge mode, they are separated by 3 spaces.
```
Comment (vpzomtrrfrt) [inline LANGUAGE.md]: This change seems irrelevant?

File: lib/parser.js [hunk 2]
```diff
@@ -116,6 +117,144 @@ function isDogescriptSource(keys)
   return false;
 }
 
+function handleWoof(keys)
+{
+
+  // woof foo is X
+  if ( keys[2] === 'is')
```
Comment (vpzomtrrfrt) [inline lib/parser.js]: Is the potential ambiguity between this and exporting an equality comparison a concern?

File: LANGUAGE.md (hunk 1 above)
Comment (AnEmortalKid) [reply LANGUAGE.md]: I noticed it's misspelled, I can revert it

File: lib/parser.js (hunk 2 above)
Comment (AnEmortalKid) [reply lib/parser.js]: I battled with this internally. originally I was using the `as` keyword like we do for require: `so blah as blah`, but that sort of shifted everything around:

```
woof such f much b wow as bar
```

Can you export an `===` ? :
```
module.exports = === b;
```

I kept `is` since at least it feels consistent (though misleading) with `very x is y`. I guess it's not really needed for the `woof` statement. Would it feel too out of place to not have it?

```
woof bar such foo
```

vs
```
woof bar is such foo
```

Comment (AnEmortalKid) [reply lib/parser.js]: Or maybe I misunderstood :( care to elaborate? @vpzomtrrfrt

**Comment by AnEmortalKid** [comment]
@vpzomtrrfrt any outstanding issues with this?

File: lib/parser.js (hunk 2 above)
Comment (vpzomtrrfrt) [reply lib/parser.js]: I think I was referring to the possibility of `module.exports = foo === bar`

Comment (vpzomtrrfrt) [reply lib/parser.js]: I probably wouldn't expect it to work in 2.x, but how would that export be written in djs?

`woof foo is bar` seems like it could be either `module.exports.foo = bar` or `module.exports = foo === bar`

Comment (AnEmortalKid) [reply lib/parser.js]: Ah... yeah I didn’t consider that. It would do `module.exports.foo = bar`

I guess if you really needed module.exports = foo === bar you could just write it in JS since we ignore non dogescript source.

Comment (AnEmortalKid) [reply lib/parser.js]: Well, drat....

would `as` instead of `is` be better? It sort of goes with the `so blah as blah2`

Comment (AnEmortalKid) [reply lib/parser.js]: @Pholey thoughts?

Comment (AnEmortalKid) [reply lib/parser.js]: So, my personal preference is to keep `is` despite ambiguity (since the usage of `so` sort of flips things around). 

@vpzomtrrfrt  or @Pholey  (any preference for one over another). I don't see an immediate use for a:

```
module.exports. = foo === bar
```

So not supporting that doesn't seem like a big deal to me

Comment (vpzomtrrfrt) [reply lib/parser.js]: While this is likely a rare construct, I'd like to eventually be able to write any non-esoteric Javascript code in native Dogescript. `as` seems like a good solution, especially with the similarity to `so`'s grammar

Comment (vpzomtrrfrt) [reply lib/parser.js]: Ambiguity causes confusion, and seems like a good thing to avoid

Comment (AnEmortalKid) [reply lib/parser.js]: We might have to make it a bit wonky:

```
woof as foo much a b
  shh 1
wow
```

Since wow is a terminating keyword it might be tricky to try to support

```
woof much a b
ahh 1
wow as foo
```

Particularly with the available tokens being provided on a line basis, we won’t have the export name until we process the `wow`

Comment (AnEmortalKid) [reply lib/parser.js]: We might have to make it a bit wonky:

```
woof as foo much a b
  shh 1
wow
```

Since wow is a terminating keyword it might be tricky to try to support

```
woof much a b
ahh 1
wow as foo
```

Particularly with the available tokens being provided on a line basis, we won’t have the export name until we process the `wow`

Comment (vpzomtrrfrt) [reply lib/parser.js]: Hmm, that's a good point

Comment (AnEmortalKid) [reply lib/parser.js]: What if we used `so` (feels odd but the woof keyword helps distinguish)

```
woof foo so much a b
  shh 1
wow
```

```
module.exports.foo = function(a,b,) { //1 }
```

Comment (AnEmortalKid) [reply lib/parser.js]: Well, other than`so` I also thought of `be`

`woof foo be much a b wow` => module.exports.foo = function (a,b)

Can’t think of any other keywords atm :(

**Comment by AnEmortalKid** [comment]
I'll push some changes with `woof x be y` as the syntax, so we can do `woof x be foo is bar`.

**Comment by AnEmortalKid** [comment]
Pushed changes with `be`. I made TODOs for the other expression supports. In a follow up PR i'm going to extract the main chunk of parse() into a `parseKeys` so all the handler functions i'm declaring outside can call that (to have more recursive power).

**Comment by AnEmortalKid** [comment]
@vpzomtrrfrt any opposition with the new keyword?

**Comment by vpzomtrrfrt** [comment]
`be` seems like the best compromise so far

File: test/spec.test.js [hunk 3]
```diff
@@ -65,36 +65,28 @@ skywalker.on('end', function() {
...
       if(skip)
       {
         t.skip('skipped');
         return;
       }
-      
+
       if (fs.existsSync(path.join(dir, 'expect.js')))
       {
-        var testFilePath   = path.join(dir, 'expect.js');
```
Comment (vpzomtrrfrt) [inline test/spec.test.js]: What is this change?  Why is it part of this branch?

Comment (AnEmortalKid) [reply test/spec.test.js]: coolios, reset!

**Comment by AnEmortalKid** [comment]
@vpzomtrrfrt ready for final blessing :)
